lighthouse-core/test/gather/gather-runner-test.js

```diff
@@ -21,6 +21,46 @@ const {createMockSendCommandFn} = require('./mock-commands.js');
 
 jest.mock('../../lib/stack-collector.js', () => () => Promise.resolve([]));
 
+/**
+ * @template {Array} TParams
+ * @template TReturn
+ * @param {(...args: TParams) => TReturn} fn
+ */
+function makeParamsOptional(fn) {
+  return /** @type {(...args: RecursivePartial<TParams>) => TReturn} */ (fn);
+}
+
+const GatherRunner = {
```

Comment:
meta question: do we still end up catching that many type bugs if we're mutating the type signature of the class under test so much?

should we be creating default configs/passes/etc to modify for each instead to maintain the integrity of the test?

Reply:
There were no type bugs that were discovered from this particular file. Perhaps other files will have some (IIRC, driver-test had one).

The larger benefit IMO is in writing new tests, modifying existing ones, or just understanding current ones in the IDA (easier with types).